Increment18-19

ncyellow opened the pull request:
18-19 инкремент
58% test coverage
--
Поправил замечания
1. Удалил по итогу залетевший левый файл
2. Удалил бесплезный тест config
3. Добавил проверку приращение счетчика. Проверять наличие всех gauges метрик не стал так как договорились что это избыточно

ncyellow committed:
Сделал multichecker.

Мультичекер со всеми требованиями инкремента

ncyellow committed:
Задание по 19 инкременту

ncyellow committed:
по требованию задания уровень тестов должен быть выше 55 - сейчас он стал 57%

ncyellow committed:
забыл прогнать стан

ncyellow committed:
добавил тесты для кастомного анализатора + для публичных анализатора

molchalin commented on internal/agent/config/config_test.go:
```diff
@@ -0,0 +1,17 @@
+)
+
+// TestConfig базовая проверка что конфиг агента в порядке
+func TestConfig(t *testing.T) {
```
Какой то странный тест. По сути мы тут проверяем возможности языка Go)

ncyellow replied:
Удалил

molchalin commented on internal/agent/metric_sources_test.go:
```diff
@@ -0,0 +1,27 @@
+
+// TestRuntimeSource проверяем что источник возвращает корректное число метрик
+// counter и gauge
+func TestRuntimeSource(t *testing.T) {
```
Тут тоже не совсем понятно, что проверяется. Можем обсудить тему тестирования на 1-1

ncyellow replied:
Тут решили добавить проверку инкремента счетчика - Сделано.
Но добавлять проверку всех Guages метрик не стали - ибо избыточно

molchalin commented on internal/server/storage/file doesn't not exists:
```diff
@@ -0,0 +1 @@
+[]
```
Какой то странный файл)

ncyellow replied:
Левый файл - удалил

molchalin reviewed (commented):
Давай немного поработаем над тестированием. Сейчас тесты проверяют результат поверхностно

ncyellow committed:
Исправления замечаний

1. Удалил по итогу залетевший левый файл
2. Удалил бесплезный тест config
3. Добавил проверку приращение счетчика. Проверять наличие всех gauges метрик не стал так как договорились что это избыточно

ncyellow committed:
Делаю единообразно
